packages/language/test/linking.test.ts

```diff
@@ -44,223 +50,253 @@ test("TODO", () => {});
+  //  end Control;
+  //  <|A|>: procedure (VAR1);
+  //  declare <|VAR1|> char(3);
+  //  put skip list(V<|>AR1);
+  //  end <|>A;`;
+
+  //       test("Must find declared procedure label in CALL", async () => {
+  //         await gotoDefinition({
+  //           text: text,
+  //           index: 0,
+  //           rangeIndex: 0,
+  //         });
+  //       });
+
+  //       test("Must find declared procedure label in END", async () => {
+  //         await gotoDefinition({
+  //           text: text,
+  //           index: 2,
+  //           rangeIndex: 0,
+  //         });
+  //       });
+
+  //       test("Must find declared variable", async () => {
+  //         await gotoDefinition({
+  //           text: text,
+  //           index: 1,
+  //           rangeIndex: 1,
+  //         });
+  //       });
+  //     });
+
+  //     describe("Qualified names", async () => {
+  //       const text = `
+  // 0DCL 1  <|TWO_DIM_TABLE|>,
+  //         2  <|TWO_DIM_TABLE_ENTRY|>               CHAR(32);
+  // 0DCL 1  TABLE_WITH_ARRAY,
+  //         2  ARRAY_ENTRY(0:1000),
+  //            3  NAME                          CHAR(32) VARYING,
+  //            3  <|TYPE#|>                         CHAR(8),
+  //         2  NON_ARRAY_ENTRY,
+  //            3  NAME                          CHAR(32) VARYING,
+  //            3  TYPE#                         CHAR(8);
+
+  //  PUT (<|>TWO_DIM_TABLE);
+  //  PUT (TWO_DIM_TABLE.<|>TWO_DIM_TABLE_ENTRY);
+  //  PUT (TABLE_WITH_ARRAY.ARRAY_ENTRY(0).<|>TYPE#);`;
+
+  //       test("Must find table name in table", async () => {
+  //         await gotoDefinition({
+  //           text: text,
+  //           index: 0,
+  //           rangeIndex: 0,
+  //         });
+  //       });
+
+  //       test("Must find qualified name in table", async () => {
+  //         await gotoDefinition({
+  //           text: text,
+  //           index: 1,
+  //           rangeIndex: 1,
+  //         });
+  //       });
+
+  //       test("Must find qualified name in array", async () => {
+  //         await gotoDefinition({
+  //           text: text,
+  //           index: 2,
+  //           rangeIndex: 2,
+  //         });
+  //       });
+  //     });
+  //   });
+
+  test("fetch linking", async () => {
+    const doc = parse(
+      `
+        MAINPR: PROCEDURE OPTIONS(MAIN);
+        dcl A entry;
+        fetch A;
+        end MAINPR;
+    `,
+      { validate: true },
+    );
+    const diagnostics = collectDiagnostics(doc);
+
+    expect(diagnostics.length).toBe(0);
+
+    const procedureStatement = doc.ast.statements[0]
+      .value as ProcedureStatement;
+    const subStmt = procedureStatement.statements[1] as Statement;
+    expect(
+      subStmt &&
+        subStmt.value &&
+        subStmt.value.kind === SyntaxKind.FetchStatement,
+    ).toBeTruthy();
+
+    // look up the symbol
+    const fetchStmt = subStmt.value as FetchStatement;
+    const fetchStmtNameRef = fetchStmt.entries[0].name as string;
```

Comment:
Suggestion: Doesn't it make more sense that the `name` property is a `ref: Reference<NamedElement>` or `call: MemberCall` instead? That way, we also get LSP support for that.

Reply:
I hadn't set that up yet after the initial rebase to the new state, but I was just coming back to this one thinking something similar as well. I can just add that on top here